prometheus/histogram.go

```diff
@@ -1835,3 +1835,143 @@ func (n *nativeExemplars) addExemplar(e *dto.Exemplar) {
 		n.exemplars = append(n.exemplars[:nIdx], append([]*dto.Exemplar{e}, append(n.exemplars[nIdx:rIdx], n.exemplars[rIdx+1:]...)...)...)
 	}
 }
+
+type constNativeHistogram struct {
+	desc *Desc
+	dto.Histogram
+	labelPairs []*dto.LabelPair
+}
+
+func validateCount(count uint64, negativeBuckets, positiveBuckets map[int]int64, zeroBucket uint64) error {
+	var bucktCountSum int64
+	for _, v := range positiveBuckets {
+		bucktCountSum += v
+	}
+	for _, v := range negativeBuckets {
+		bucktCountSum += v
+	}
+	bucktCountSum += int64(zeroBucket)
+	if bucktCountSum != int64(count) {
+		return fmt.Errorf("bucket sums do not add up to total count")
+	}
+	return nil
+}
+
+func NewConstNativeHistogram(
+	desc *Desc,
+	count uint64,
+	sum float64,
+	positiveBuckets,
+	negativeBuckets map[int]int64,
+	zeroBucket uint64,
+	nativeHistogramSchema int32,
+	nativeHistogramZeroThreshold float64,
+	createdTimestamp time.Time,
+	nativeExemplars []*dto.Exemplar,
+	labelValues ...string,
+) (Metric, error) {
+	if desc.err != nil {
+		return nil, desc.err
+	}
+	if err := validateLabelValues(labelValues, len(desc.variableLabels.names)); err != nil {
+		return nil, err
+	}
+	if nativeHistogramSchema > 8 || nativeHistogramSchema < -4 {
```

Comment:
I'm wondering if we should finally create constants for the max and min schema. This PR adds yet another occurrence.